lib/collectionspace/converter/core/acquisition.rb

```diff
@@ -9,73 +9,100 @@ def convert
+            'acquisitionsourceperson' => ['acquisitionSources', 'acquisitionSource'],
+            'acquisitionsourceorganization' => ['acquisitionSources', 'acquisitionSource'],
+            'fieldcollectioneventname' => ['fieldCollectionEventNames', 'fieldCollectionEventName']
+          }
+          repeatstransforms = {
+            'acquisitionsourceperson' => {'authority' => ['personauthorities', 'person']},
+            'acquisitionsourceorganization' => {'authority' => ['orgauthorities', 'organization']},
+            'ownerperson' => {'authority' => ['personauthorities', 'person']},
+            'ownerorganization' => {'authority' => ['orgauthorities', 'organization']}
+          }
+          CSXML::Helpers.add_repeats(xml, attributes, repeats, repeatstransforms)
+          
+          CSXML::Helpers.add_date_group(xml, 'accessionDate', CSDTP.parse(attributes['accessiondate']))
 
-          object_purchase_offer_currency = CSXML::Helpers.get_vocab('currency', attributes['objectpurchaseofferpricecurrency'])
-          CSXML.add xml, 'objectPurchaseOfferPriceCurrency', object_purchase_offer_currency
-          CSXML.add xml, 'objectPurchaseOfferPriceValue', attributes['objectpurchaseofferpricevalue']
+          CSXML::Helpers.add_date_group_list(xml, 'acquisitionDate', attributes['acquisitiondate'])
 
-          object_purchase_currency = CSXML::Helpers.get_vocab('currency', attributes['objectpurchasepricecurrency'])
-          CSXML.add xml, 'objectPurchasePriceCurrency', object_purchase_currency
-          CSXML.add xml, 'objectPurchasePriceValue', attributes['objectpurchasepricevalue']
 
-          original_object_currency = CSXML::Helpers.get_vocab('currency', attributes['originalobjectpurchasepricecurrency'])
-          CSXML.add xml, 'originalObjectPurchasePriceCurrency', original_object_currency
-          CSXML.add xml, 'originalObjectPurchasePriceValue', attributes['originalobjectpurchasepricevalue']
+          funding_data = {
+            'acquisitionfundingcurrency' => 'acquisitionFundingCurrency',
+            'acquisitionfundingvalue' => 'acquisitionFundingValue',
+            'acquisitionfundingsourceorganization' => 'acquisitionFundingSource',
+            'acquisitionfundingsourceperson' => 'acquisitionFundingSource',
+            'acquisitionfundingsourceprovisos' => 'acquisitionFundingSourceProvisos',
+          }
+          funding_transforms = {
+            'acquisitionfundingcurrency' => {'vocab' => 'currency'},
+            'acquisitionfundingsourceorganization' => {'authority' => ['orgauthorities', 'organization']},
+            'acquisitionfundingsourceperson' => {'authority' => ['personauthorities', 'person']}
+          }
+          CSXML.add_single_level_group_list(
+            xml, attributes,
+            'acquisitionFunding',
+            funding_data,
+            funding_transforms,
+            list_suffix: 'List',
+            group_suffix: ''
+          )
```

Comment:
@Paulmulonzia Prefer to use `add_single_level_group_list` rather than `add_list`.

The following are both single level field groupings: 

```
<acquisitionFundingList>                         <approvalGroupList>
  <acquisitionFunding>                                  <approvalGroup>
     <acquisitionFundingSource/>                           <approvalStatus/>
     <acquisitionFundingCurrency/>                         <approvalNote/>
  </acquisitionFunding>                                 </approvalGroup>
</acquisitionFundingList>                        </approvalGroupList>
``` 

If you do not specify suffix values when calling `add_single_level_group_list`, it will add "GroupList" to the top level and "Group" to the next level by default. (as on the right)

By specifying "List" as the list-level suffix and "" as the group level suffix here, we get the needed labels. 

*What I'm going for is one method for each field structure pattern,* not a bunch of different methods to handle variations of field naming conventions.

Reply:
Sticking to a single approach sounds good to me.